### PR TITLE
Yosys - ISE flow fix

### PR DESCRIPTION
**Pull request by mmicko**
These changes are needed due to recent better support introduced in Yosys specially regarding Spartan6 flow.

Tested this with https://github.com/mmicko/linux-on-litex-vexriscv/tree/spartan6 for creating Linux running on WaxWing Spartan6 board from Numato

### CODE REVIEW

File: litex/build/xilinx/ise.py [hunk 1]
```diff
@@ -207,7 +218,8 @@ def build(self, platform, fragment, build_dir="build", build_name="top",
                     _build_xst_files(platform.device, platform.sources, platform.verilog_include_paths, build_name, self.xst_opt)
                     isemode = mode
                 else:
-                    _run_yosys(platform.device, platform.sources, platform.verilog_include_paths, build_name)
+                    if run:
```
Comment (mithro) [inline litex/build/xilinx/ise.py]: Where does `run` come from?

Comment (mithro) [reply litex/build/xilinx/ise.py]: Oh wait, it is an argument....

File: litex/build/xilinx/ise.py [hunk 2]
```diff
@@ -114,7 +122,10 @@ def _run_ise(build_name, ise_path, source, mode, ngdbuild_opt,
         settings = common.settings(ise_path, ver, "ISE_DS")
         build_script_contents += source_cmd + tools.cygpath(settings) + "\n"
     if mode == "edif":
-        ext = "edif"
+        ext = "ngo"
```
Comment (mithro) [inline litex/build/xilinx/ise.py]: Why is this needed now? I'm a little worried it will break other edif use cases?

File: litex/build/xilinx/ise.py (hunk 1 above)
Comment (mmicko) [reply litex/build/xilinx/ise.py]: Right. This is added since it executes yosys right now even when you do not build but load on board

File: litex/build/xilinx/ise.py (hunk 2 above)
Comment (mmicko) [reply litex/build/xilinx/ise.py]: ngo and ngc are supported by ngdbuild and other commands in flow. Adding edif2ngd is cleaner way to do the flow instead of using edif directly (Think for actually did not work fine at all for me , but away right now from developer machine, so can confirm tomorrow)

Comment (mmicko) [reply litex/build/xilinx/ise.py]: Just tested. Was right, this was broken before. Without change it fails at this line :

netgen -ofmt verilog -w -sim top.edif top_synth.v

Release 14.7 - netgen P.20131013 (lin64)
Copyright (c) 1995-2013 Xilinx, Inc.  All rights reserved.
ERROR:NetListWriters:286 - Invalid file argument 'top.edif'.

Thing is netgen works with ngc or ngo, and edif was sent, that is why I add step to convert edif to ngc and made this work actually.

Comment (mithro) [reply litex/build/xilinx/ise.py]: Great!

File: litex/build/xilinx/ise.py (hunk 1 above)
Comment (mithro) [reply litex/build/xilinx/ise.py]: Great!

**Comment by enjoy-digital** [comment]
Thanks @mmicko and @mithro.
